sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVFileFormat.scala

```diff
@@ -86,7 +86,7 @@ class CSVFileFormat extends TextBasedFileFormat with DataSourceRegister {
       }
 
       override def getFileExtension(context: TaskAttemptContext): String = {
-        ".csv" + CodecStreams.getCompressionExtension(context)
+        "." + csvOptions.fileExtension + CodecStreams.getCompressionExtension(context)
```

Comment:
In this way, the user can overwrite almost anything in the file system:
```
./../../../../../path/<file to overwrite>.xz
```
Should we do some sanity check of the option?

Reply:
That seems reasonable to me. I can add a check that limits the file extension to three characters that are limited to a-z. If other characters are found we can throw an exception stating the limitation.

Reply:
I have added a limit of 3 letters when we set the option.

See: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala